vm/map.c

```diff
@@ -1003,6 +1005,48 @@ void vm_mapinfo(meminfo_t *info)
 		proc_lockClear(&map_common.kmap->lock);
 		info->entry.kmapsz = size;
 	}
+
+	if (info->maps.mapsz != -1) {
+		info->maps.total = 0;
+		info->maps.free = 0;
+
+		for (size = 0; size < map_common.mapssz; ++size) {
+			map = vm_getSharedMap(NULL, (int)size);
+			if (map == NULL) {
+				if ((int)size < info->maps.mapsz) {
```

Comment:
If `size >= info->maps.mapsz`, there is nothing to do and You can exit the loop.
If You accept this suggestion, You can remove `if` in line `1016` and `1037`.

```suggestion
        for (size = 0; size < map_common.mapssz; ++size) {
                if ((int)size >= info->maps.mapsz) {
                        break;
                }
		map = vm_getSharedMap(NULL, (int)size);
```

Reply:
Nope, there is something to do - e.g. use case with info->map.mapsz = 0 exist - if I only want total summary, without additional information about specific maps.